pkg/readiness/objset.go

```diff
@@ -27,9 +27,35 @@ type objKey struct {
 	namespacedName types.NamespacedName
 }
 
-type objSet map[objKey]struct{}
-
 func (k objKey) String() string {
 	return fmt.Sprintf("%s [%s]", k.namespacedName.String(), k.gvk.String())
+}
+
+type objData struct {
+	retries int
+}
 
+// decrementRetries handles objData retries, and returns whether it's time to delete the objData.
```

Comment:
nit: "returns `true` if it's time to delete the objData entry"

disambiguate the meaning of the return value